paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java

```diff
@@ -161,6 +161,18 @@ public Optional<MetastoreClient.Factory> metastoreClientFactory(Identifier ident
         }
     }
 
+    @Override
+    public Path getDataTableLocation(Identifier identifier) {
+        try {
+            return new Path(
+                    client.getTable(identifier.getDatabaseName(), identifier.getObjectName())
+                            .getSd()
+                            .getLocation());
+        } catch (TException e) {
```

Comment:
```
catch (NoSuchObjectException e) {
     return super.getDataTableLocation(identifier);
}
```

Reply:
Regarding HiveCatalog, I have a question: should we fallback to the parent implemets, or should we adhere to the Hive specification and create tables under the database path? @JingsongLi

Reply:
OK let's just throw exception.